### PR TITLE
Connections update

### PR DESCRIPTION
**Pull request by nbildiri**

### CODE REVIEW

File: src/main/java/Connections.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        ArrayList<Station> connections = new ArrayList<>();
+        connections.add(station);
+        Station currStation;
+
+        for (Line line : lines) {
+            for (int index = 0; index < line.getStops().size(); index++) {
+                if (line.getStops().get(index) == station.getProperties().getObjectid()) {
+
+                    //if not first in list
+                    if (!Objects.equals(line.getStops().get(index), line.getStops().get(0)))
```
Comment (andrewoid) [inline src/main/java/Connections.java:23]: `!line.getStops().get(index).equals(line.getStops().get(0)))`

Comment (andrewoid) [inline src/main/java/Connections.java:23]: `if (index != 0)`

File: src/main/java/Connections.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }*/
+
+    public Station getClosestStation(AllStations stations, double lat, double lon) {
+
+        double closestDist = Integer.MAX_VALUE;
+        double dist;
+        Station closestStation = null;
+
+        for (Station station : stations.getStations()) {
+            dist = Math.sqrt((Math.pow((station.getGeometry().getLatitude() - lat), 2)) +
```
Comment (andrewoid) [inline src/main/java/Connections.java:64]: make a distance method that you can reuse on line 78

File: src/main/java/Connections.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    public double getDistanceBetweenStations(Station station_1, Station station_2) {
+
+        double dist = Math.sqrt((Math.pow((station_2.getGeometry().getLatitude() -
+                station_1.getGeometry().getLatitude()), 2)) +
+                (Math.pow((station_2.getGeometry().getLongitude() -
+                        station_2.getGeometry().getLongitude()), 2)));
+
+        return dist;
```
Comment (andrewoid) [inline src/main/java/Connections.java:83]: you can put this return on line 78.

File: src/main/java/ParseJson.java [hunk 4]
```diff
@@ -3,11 +3,12 @@
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.sql.CallableStatement;
+import java.util.HashMap;
 
 import com.google.gson.*;
 
 
-public class ParseJson {
+public class ParseJson extends HashMap{
```
Comment (andrewoid) [inline src/main/java/ParseJson.java:11]: This class should not extend HashMap

File: src/main/java/ParseLine.java [hunk 5]
```diff
@@ -3,37 +3,36 @@
...
+public class ParseLine{
 
-   public ArrayList<Line> parseLine() throws IOException {
+    public ArrayList<Line> parseLine() throws IOException {
 
         ArrayList<Line> lines = new ArrayList<>();
 
-            // create Gson instance
-            Gson gson = new Gson();
+        // create Gson instance
```
Comment (andrewoid) [inline src/main/java/ParseLine.java:19]: You can remove useless comments like this

File: src/main/java/Properties.java [hunk 6]
```diff
@@ -1,26 +1,29 @@
+import java.util.Arrays;
+import java.util.List;
+
 public class Properties {
 
-    private String objectid;
+    private Integer objectid;
```
Comment (andrewoid) [inline src/main/java/Properties.java:6]: `int`

File: src/test/java/ParseJsonTest.java [hunk 7]
```diff
@@ -1,54 +1,104 @@
...
         //stop
-        assertEquals("55", lines.get(0).getStops().get(0));
+        assertEquals(55, lines.get(0).getStops().get(0));
+
+
+    }
+
+    @Test
+    void getConnectionsTest() throws IOException {
+        //given
```
Comment (andrewoid) [inline src/test/java/ParseJsonTest.java:60]: it's given-when-then

File: src/test/java/ParseJsonTest.java [hunk 8]
```diff
@@ -1,54 +1,104 @@
...
 
         //stop
-        assertEquals("55", lines.get(0).getStops().get(0));
+        assertEquals(55, lines.get(0).getStops().get(0));
+
+
+    }
+
+    @Test
+    void getConnectionsTest() throws IOException {
```
Comment (andrewoid) [inline src/test/java/ParseJsonTest.java:59]: `getConnections()`

File: src/main/java/Connections.java [hunk 9]
```diff
@@ -20,67 +19,45 @@
...
     public double getDistanceBetweenStations(Station station_1, Station station_2) {
 
-        double dist = Math.sqrt((Math.pow((station_2.getGeometry().getLatitude() -
-                station_1.getGeometry().getLatitude()), 2)) +
-                (Math.pow((station_2.getGeometry().getLongitude() -
-                        station_2.getGeometry().getLongitude()), 2)));
+        return distance(station_1, station_2.getGeometry().getLatitude(), station_2.getGeometry().getLongitude());
+    }
+
+    private double distance(Station station, double lat, double lon) {
```
Comment (andrewoid) [inline src/main/java/Connections.java:57]: now create a second method that takes 2 Stations so you can clean up line 54

File: src/main/java/Connections.java [hunk 10]
```diff
@@ -20,67 +19,45 @@
...
                 closestStation = station;
                 closestDist = dist;
             }
         }
-
         return closestStation;
     }
 
+
     public double getDistanceBetweenStations(Station station_1, Station station_2) {
```
Comment (andrewoid) [inline src/main/java/Connections.java:52]: variable names should not have _s

File: src/main/java/Properties.java [hunk 11]
```diff
@@ -1,26 +1,29 @@
...
+
 public class Properties {
 
-    private String objectid;
+    private int objectid;
     private String name;
     private String line;
 
-    public Properties(String objectid, String name, String line) {
+    public Properties(Integer objectid, String name, String line) {
```
Comment (andrewoid) [inline src/main/java/Properties.java:10]: `int`

File: src/main/java/Connections.java (hunk 9 above)
Comment (nbildiri) [reply src/main/java/Connections.java:57]: but I thought the instructions were that we would receive coordinates and from there we have to find the closest station and shortest path

File: src/main/java/Connections.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    private double distance(Station station, double lat, double lon) {
+
+        return Math.sqrt((Math.pow((station.getGeometry().getLatitude() - lat), 2)) +
+                (Math.pow((station.getGeometry().getLongitude() - lon), 2)));
+    }
+
+    /*
+    public void stationConnections(AllStations stations, double lat_1, double lon_1, double lat_2, double lon_2) {
```
Comment (andrewoid) [inline src/main/java/Connections.java:64]: Why is this commented out?

File: src/main/java/Geometry.java [hunk 13]
```diff
@@ -1,19 +1,23 @@
...
 
-    public double[] getCoordinates() {
+    public ArrayList<Double> getCoordinates() {
         return coordinates;
     }
 
-    public double getCoordinateInArray(int num) {
-        return coordinates[num];
+    public double getLatitude() {
+        return getCoordinates().get(0);
```
Comment (andrewoid) [inline src/main/java/Geometry.java:16]: you can just do
`coordinates.get(0)`

But you could've left this an an ArrayList and not changed it.

File: src/main/java/Graph.java [hunk 14]
```diff
@@ -0,0 +1,91 @@
+/*
+import java.util.ArrayList;
+
+public class Graph {
+
+
+/*
+  Let the node at which we are starting at be called the initial node. Let the distance of node Y be the distance from the initial node to Y. Dijkstra's algorithm will initially start with infinite distances and will try to improve them step by step.
```
Comment (andrewoid) [inline src/main/java/Graph.java:8]: Why is this commented out?

File: src/test/java/ParseJsonTest.java [hunk 15]
```diff
@@ -1,54 +1,104 @@
...
+                40.73005400028978);
+
+        assertEquals(stations.getStations().get(0).getProperties().getName(), closestStation.getProperties().getName());
+
+
+        //end point - 238th street
+        closestStation = connections.getClosestStation(stations, -73.90087000018522,
+                40.88466700064975);
+
+        assertEquals(stations.getStations().get(5).getProperties().getName(), closestStation.getProperties().getName());
```
Comment (andrewoid) [inline src/test/java/ParseJsonTest.java:93]: This should be a separate test method.

**Pull request closed without merging**